overlord/restart/restart.go

```diff
@@ -217,10 +217,19 @@ func (rm *RestartManager) PendingForSystemRestart(chg *state.Change) bool {
 			continue
 		}
 
-		if rm.bootID == waitBootId {
-			// no boot has intervened yet
+		if rm.bootID != waitBootId {
+			// this should not happen as it
+			// means StartUp did not operate correctly,
+			// but if it happens fair game for aborting
 			continue
 		}
+		// no boot intervened yet
+		// no successive tasks, take the WaitStatus at face value
+		if len(t.HaltTasks()) == 0 {
+			return true
```

Comment:
Nitpick: might be more clear with some spacing and moving comments around:
```suggestion
		// no boot intervened yet

		if len(t.HaltTasks()) == 0 {
			// no successive tasks, take the WaitStatus at face value
			return true
```